config/202-gateway.yaml

```diff
@@ -37,70 +37,106 @@ spec:
     tls:
       mode: PASSTHROUGH
 ---
+# TODO(#1969): We should allow the users to use `istio-ingressgateway` by default,
+#              while having the choice to specify their own Ingress Gateway service
+#              if that isn't enough for them.  That way we can get out of maintaining
+#              these YAML ourselves.
+#
 # This is the Service definition for the ingress pods serving
 # Knative's shared Gateway.
-#
-# Source: istio/charts/ingressgateway/templates/service.yaml
 apiVersion: v1
 kind: Service
 metadata:
   name: knative-ingressgateway
   namespace: istio-system
+  annotations:
   labels:
-    chart: ingressgateway-0.8.0
+    chart: gateways-1.0.1
     release: RELEASE-NAME
     heritage: Tiller
+    app: knative-ingressgateway
     knative: ingressgateway
 spec:
   type: LoadBalancer
   selector:
+    app: knative-ingressgateway
     knative: ingressgateway
   ports:
     -
-      name: http
+      name: http2
       nodePort: 32380
       port: 80
+      targetPort: 80
     -
       name: https
       nodePort: 32390
       port: 443
     -
       name: tcp
       nodePort: 32400
-      port: 32400
+      port: 31400
+    -
+      name: tcp-pilot-grpc-tls
+      port: 15011
+      targetPort: 15011
+    -
+      name: tcp-citadel-grpc-tls
+      port: 8060
+      targetPort: 8060
+    -
+      name: tcp-dns-tls
+      port: 853
+      targetPort: 853
+    -
+      name: http2-prometheus
+      port: 15030
+      targetPort: 15030
+    -
+      name: http2-grafana
+      port: 15031
+      targetPort: 15031
 ---
 # This is the corresponding Deployment to backed the aforementioned Service.
-#
-# Source: istio/charts/ingressgateway/templates/deployment.yaml
```

Comment:
Why did these `Source` comment lines go away? What is the process for doing this again if we discover (for example) that istio-1.0.4 is more stable than 1.0.1?

Reply:
Good suggestion!  I will add more comments/README.md in the master branch and update to this PR.